src/app/drive/components/ItemDetailsDialog/ItemDetailsDialog.tsx

```diff
@@ -0,0 +1,198 @@
+import { useEffect, useState } from 'react';
+
+import { RootState } from '../../../store';
+import { useAppDispatch, useAppSelector } from '../../../store/hooks';
+import { uiActions } from '../../../store/slices/ui';
+import Modal from '../../../shared/components/Modal';
+import { X } from '@phosphor-icons/react';
+import { useTranslationContext } from '../../../i18n/provider/TranslationProvider';
+import iconService from '../../../drive/services/icon.service';
+import Button from '../../../shared/components/Button/Button';
+import { bytesToString } from '../../../drive/services/size.service';
+import date from '../../../core/services/date.service';
+import localStorageService from '../../../core/services/local-storage.service';
+import useDriveItemActions from '../DriveExplorer/DriveExplorerItem/hooks/useDriveItemActions';
+import { DriveItemData, DriveItemDetails, ItemDetailsProps } from '../../../drive/types';
+import newStorageService from 'app/drive/services/new-storage.service';
+import errorService from 'app/core/services/error.service';
+import { getItemPlainName } from 'app/crypto/services/utils';
+import ItemDetailsSkeleton from './components/ItemDetailsSkeleton';
+
+const Header = ({ title, onClose }: { title: string; onClose: () => void }) => {
+  return (
+    <>
+      <span
+        className="max-w-full overflow-hidden overflow-ellipsis whitespace-nowrap text-xl font-medium"
+        title={title}
+      >
+        {title}
+      </span>
+      <div className="flex h-9 w-9 cursor-pointer items-center justify-center rounded-md bg-black bg-opacity-0 transition-all duration-200 ease-in-out hover:bg-opacity-4 active:bg-opacity-8">
+        <X onClick={onClose} size={22} />
+      </div>
+    </>
+  );
+};
+
+const ItemsDetails = ({ item, translate }: { item: ItemDetailsProps; translate: (key: string) => string }) => {
+  return (
+    <>
+      {Object.entries(item).map(([key, value]) => {
+        if (!value) return;
+        return (
+          <div key={key} className="flex w-full max-w-xxxs flex-col items-start justify-center space-y-0.5">
+            <p className="text-sm font-medium text-gray-50">
+              {translate(`modals.itemDetailsModal.itemDetails.${key}`)}
+            </p>
+            <p title={value} className="w-full text-base font-medium text-gray-100 line-clamp-2">
+              {value}
+            </p>
+          </div>
+        );
+      })}
+    </>
+  );
+};
+
+/**
+ * Return all the details of the item selected
+ * The data is:
+ * - Name
+ * - Shared
+ * - Size (only for files)
+ * - Type (only for files)
+ * - Uploaded
+ * - Modified
+ * - Uploaded by
+ * - Location
+ *  */
+
+const ItemDetailsDialog = ({ onSharedFolderClicked }: { onSharedFolderClicked?: (item: any) => void }) => {
```

Comment:
I use `any` instead of `item as unknown as AdvancedSharedItem` when passing the item through the function, since the item we pass does not have the same attributes as the object expected in the function.